src/satosa/backends/saml2.py

```diff
@@ -50,27 +50,23 @@ def get_memorized_idp(context, config, force_authn):
     return value
 
 
-def get_force_authn(context, config, sp_config):
+def get_saml_param(context, config, sp_config, key):
     """
-    Return the force_authn value.
+    Return the value of ForceAuthn or IsPassive.
 
     The value comes from one of three place:
     - the configuration of the backend
     - the context, as it came through in the AuthnRequest handled by the frontend.
-      note: the frontend should have been set to mirror the force_authn value.
+      note: the frontend should have been set to mirror the value.
     - the cookie, as it has been stored by the proxy on a redirect to the DS
-      note: the frontend should have been set to mirror the force_authn value.
+      note: the frontend should have been set to mirror the value.
 
     The value is either "true" or None
     """
-    mirror = config.get(SAMLBackend.KEY_MIRROR_FORCE_AUTHN)
-    from_state = mirror and context.state.get(Context.KEY_FORCE_AUTHN)
-    from_context = (
-        mirror and context.get_decoration(Context.KEY_FORCE_AUTHN) in ["true", "1"]
```

Comment:
Keep in mind that "1" is a valid value for booleans for XML. And it is used in practice.

Reply:
restricted to `"true"` and `"1"`